git.md

```diff
@@ -143,7 +143,10 @@ push *all* remote-tracking branches.
 
 So, if you ever need to force push the 'foobar' branch, please instead do
 
-    $ git push -f origin foobar
+    $ git push --force-with-lease origin foobar
+
+`--force-with-lease` refuses to update a branch unless it is the state that we expect; i.e. nobody has updated the branch upstream.
```

Comment:
"updated the branch upstream" might be clearer if you replace "upstream" with "origin", or "remote".

I'm used to using upstream in the context of free software projects, and not Git remotes.